packages/firestore/test/integration/api/transactions.test.ts

```diff
@@ -23,175 +23,182 @@ import * as integrationHelpers from '../util/helpers';
+      await this.runTransaction()
+        .then(async () => {
+          const snapshot = await this.docRef.get();
+          expect(snapshot.exists).to.equal(true);
+          expect(snapshot.data()).to.deep.equal(expected);
         })
-        .then(async (docRef: firestore.DocumentReference) => {
-          const snapshot = await docRef.get();
-          if (expected) {
-            expect(snapshot.exists).to.equal(true);
-            expect(snapshot.data()).to.deep.equal(expected);
-          } else {
-            expect(snapshot.exists).to.equal(false);
-          }
+        .catch(err => {
+          expect.fail(
+            'Expected the sequence (' +
+              this.listSteps(this.stages) +
+              ') to succeed, but got ' +
+              err
+          );
+        });
+      this.cleanupTester();
+    }
+
+    async expectNoDoc(): Promise<void> {
+      await this.prepareDoc();
+      await this.runTransaction()
+        .then(async () => {
+          const snapshot = await this.docRef.get();
+          expect(snapshot.exists).to.equal(false);
         })
         .catch(err => {
           expect.fail(
             'Expected the sequence (' +
-              this.listSteps(stages) +
+              this.listSteps(this.stages) +
               ') to succeed, but got ' +
               err
           );
         });
+      this.cleanupTester();
+    }
+
+    async expectError(expected: string): Promise<void> {
+      await this.prepareDoc();
+      await this.runTransaction().then(
+        () => {
+          expect.fail(
+            'Expected the sequence (' +
+              this.listSteps(this.stages) +
+              ') to fail with the error ' +
+              expected
+          );
+        },
+        err => {
+          expect((err as firestore.FirestoreError).code).to.equal(expected);
+        }
+      );
+      this.cleanupTester();
     }
 
-    /**
-     * Validates that executing the provided steps in a transaction results a
-     * FirestoreError after executing the transaction.
-     *
-     * @param expected The expected FirestoreError code.
-     * @param docExists Whether the document the transaction is being run on
-     * should exist beforehand.
-     * @param stages The series of transaction steps to execute inside the
-     * transaction.
-     */
-    async validateSequenceError(
-      expected: string,
-      docExists: boolean,
-      ...stages: TransactionStep[]
-    ) {
-      const docRef = this.db.collection('sighs').doc();
-      if (docExists) {
-        await docRef.set({ foo: 'bar0' });
-        const docSnap = await docRef.get();
+    private async prepareDoc(): Promise<void> {
+      this.docRef = this.db.collection('tester-docref').doc();
+      if (this.fromExistingDoc) {
+        await this.docRef.set({ foo: 'bar0' });
+        const docSnap = await this.docRef.get();
         expect(docSnap.exists).to.equal(true);
+      } else {
+        const docSnap = await this.docRef.get();
```

Comment:
This effectively guaranteed true by creating a random document ID. You can drop this check and speed things up.